src/Infrastructure/Identity/RoleService.cs

```diff
@@ -153,9 +162,15 @@ public async Task<string> RegisterRoleAsync(RoleRequest request)
         }
     }
 
-    public async Task<string> UpdatePermissionsAsync(string roleId, List<UpdatePermissionsRequest> permissions, CancellationToken cancellationToken)
+    /// <summary>
+    /// Update Permissions by Role Async.
+    /// </summary>
+    /// <param name="roleId"></param>
+    /// <param name="selectedPermissions"></param>
+    /// <param name="cancellationToken"></param>
+    /// <returns></returns>
+    public async Task<string> UpdatePermissionsAsync(string roleId, List<UpdatePermissionsRequest> selectedPermissions, CancellationToken cancellationToken)
```

Comment:
Sorry to keep bothering you... but this still doesn't feel right... I know it's originally not your code... but while we're at it... and in the spirit of applying some consistency throughout the code base:

I think the method should take in only one "UpdatePermissionsRequest", not a list and not the roleId, and the UpdatePermissionsRequest class should contain the roleId and a list of permissions (can be just a list of strings no need for another dto). Similar like we have with the RoleRequest and the RegisterRoleAsync method...

Reply:
No Problem , Ok, will update that request

Reply:
Can you review again please, 
Thanks and sorry for so many fixs, but i really thought that the original code was already improved :-p